saleor/graphql/shipping/utils.py

```diff
@@ -0,0 +1,77 @@
+from typing import TYPE_CHECKING, List, Optional
+
+from ...account.models import Address
+from ...plugins.base_plugin import ExcludedShippingMethod
+from ...plugins.base_plugin import ShippingMethod as ShippingMethodDataclass
+from ...shipping import models as shipping_models
+from ..channel import ChannelContext
+
+if TYPE_CHECKING:
+    from ...plugins.manager import PluginsManager
+    from ...shipping.models import ShippingMethodChannelListing
+
+
+def convert_shipping_method_model_to_dataclass(
+    shipping_method: shipping_models.ShippingMethod,
+):
+    shipping_method_dataclass = ShippingMethodDataclass(
+        id=str(shipping_method.id),
+        price=shipping_method.price,  # type: ignore
+        name=shipping_method.name,
+        maximum_delivery_days=shipping_method.maximum_delivery_days,
+        minimum_delivery_days=shipping_method.minimum_delivery_days,
+        maximum_order_weight=None,
+        minimum_order_weight=None,
+    )
+    shipping_method_dataclass.maximum_order_weight = (
+        shipping_method.maximum_order_weight
+    )
+    shipping_method_dataclass.minimum_order_weight = (
+        shipping_method.minimum_order_weight
+    )
+    return shipping_method_dataclass
+
+
+def annotate_shipping_methods_with_price(
+    shipping_methods: List[shipping_models.ShippingMethod],
+    channel_listings: List["ShippingMethodChannelListing"],
+    address: Optional["Address"],
+    channel_slug: str,
+    manager: "PluginsManager",
+):
+    if not address:
+        return
+    channel_listing_map = {
+        channel_listing.shipping_method_id: channel_listing
+        for channel_listing in channel_listings
+    }
+    for method in shipping_methods:
+        shipping_channel_listing = channel_listing_map[method.id]
+        method.price = shipping_channel_listing.price  # type: ignore
+
+
+def annotate_active_shipping_methods(
+    shipping_methods: List[shipping_models.ShippingMethod],
+    excluded_methods: List[ExcludedShippingMethod],
+):
+    for instance in shipping_methods:
+        instance.active = True  # type: ignore
+        instance.message = ""  # type: ignore
+        for method in excluded_methods:
+            if str(instance.id) == str(method.id):
+                instance.active = False  # type: ignore
+                instance.message = method.reason  # type: ignore
```

Comment:
@stnatic Assigning fields to an instance on the fly is not a practice that we do in Saleor - this will have to be refactored later - please make sure that you have it written down in a task with things to refactor. CC @patrys just to make sure that you saw this.

Reply:
@maarcingebala This is already changed on 3.1 where all delivery method roots are instances of the ShippingMethodData. I agree that it would be a good idea backport some of that functionality to 3.0.

Please note that Saleor has been already doing this to add fields of `ShippingMethodChannelListing` to a `ShippingMethod` when displaying it in a context of a specific channel, so we didn't introduce a new convention here:
https://github.com/saleor/saleor/pull/8399/files#diff-473baa60c7b2166e11d93921bf9fee5243a14adb9198e5ddc452d1cc8776ec6cL407

Reply:
It doesn't change the fact that this is a bad convention and should be avoided; we should probably change the place you linked as well.

Reply:
Fully agreed! My recommendation would be to first approve the solution on 3.1 and backport the `ShippingMethodData` interface onto 3.0 as well. This way we will make sure that the shipping logic in both doesn't diverge even further

Reply:
As we discussed a couple of times. The main priority is to finish the changes from 3.0, and apply the requested changes from the previous PRs which we converted to task.  We don't think about 3.1 yet as we have limited time to finish 3.0.